src/sink.cc

```diff
@@ -164,9 +165,36 @@ void Sink::Tick() {
   using std::vector;
   LOG(cyclus::LEV_INFO3, "SnkFac") << prototype() << " is ticking {";
 
-  double requestAmt = RequestAmt();
+  double amt = RequestAmt();
+  double requestAmt = 0;
+
+  LOG(cyclus::LEV_INFO3, "SnkFac") << prototype() << " has default request amount " << amt;
+
   // inform the simulation about what the sink facility will be requesting
-  if (requestAmt > cyclus::eps()) {
+  if (amt > cyclus::eps()) {
+    if (random_size == "None") {
+      requestAmt = amt;
+    }
+    else if (random_size == "UniformInt") {
+      requestAmt = context()->random_uniform_int(0, amt);
+    }
+    else if (random_size == "UniformReal") {
+      requestAmt = context()->random_uniform_real(0, amt);
+    }
+    else if (random_size == "NormalReal") {
+      requestAmt = context()->random_normal_real(amt * random_size_mean,
+      amt * random_size_stddev, 0, amt);
+    }
+    else if (random_size == "NormalInt") {
+      requestAmt = context()->random_normal_int(amt * random_size_mean,
+      amt * random_size_stddev, 0, amt);
+    }
+    else {
+      requestAmt = amt;
```

Comment:
Since "None" is the default, should this throw an exception/error?  XML validation should prevent it from happening at all.

Reply:
Things got moved around in the latest PR, but I left this in. The main reason is testing-- some tests don't fully spin up a simulation, which results in default values not being assigned (i.e. left uninitialized). Without a catch-all else, this code* also results in an uninitialized `requestAmt` in some tests.

. This catches that particular case and ensures that `SpaceAvailable()` is used in that case. I know it's inelegant, happy to take other suggestions

*including in in its new iteration as `SetRequestAmt()`